src/main/kotlin/com/bridgecrew/ui/vulnerabilitiesTree/CheckovToolWindowTree.kt

```diff
@@ -82,17 +92,73 @@ class CheckovToolWindowTree(val project: Project, val split: JBSplitter, private
             }
         }
 
+        expandsFromPathState(tree, CheckovGlobalState!!.expandedDescendants)
+
+        tree.addTreeExpansionListener(object : TreeExpansionListener {
+            override fun treeExpanded(event: TreeExpansionEvent?) {
+                CheckovGlobalState!!.expandedDescendants = getExpandedDescendants((event?.source as Tree))
+            }
+
+            override fun treeCollapsed(event: TreeExpansionEvent?) {
+                CheckovGlobalState!!.expandedDescendants = getExpandedDescendants((event?.source as Tree))
+            }
+        })
+
         resultsPanel.add(tree)
         isTreeEmpty = tree.isEmpty
         return resultsPanel
     }
 
-    private fun createFolderTree(currentTree: DefaultMutableTreeNode, resultsPerFile: List<BaseCheckovResult>, fileName: String) {
+    private fun expandsFromPathState(tree: Tree, expandsTreePaths: List<TreePath>) {
+        expandsTreePaths.forEach {
+//            TODO: instead of searching the tree for each, expand while scanning
+                path ->
+            tree.expandPath(
+                findTreePathByUserObject(
```

Comment:
maybe it's a personal preference but I would prefer storing the node first, then using it, for readability

Reply:
Also maybe necessary because we don't account for the case the node isn't found (worth also checking it's not null)